sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityGenerator.scala

```diff
@@ -95,7 +95,7 @@ class UnivocityGenerator(
       if (!row.isNullAt(i)) {
         values(i) = valueConverters(i).apply(row, i)
       } else {
-        values(i) = options.nullValue
+        values(i) = null
```

Comment:
I see. So Univocity will write out the values for actual `null`s`. I think we can even just don't explicitly set `null`s `here`. e.g.)

```scala
      if (!row.isNullAt(i)) {
        values(i) = valueConverters(i).apply(row, i)
      }
```

Reply:
Yes! That is better and `values(i)` is null originally.